### PR TITLE
fix(argo-cd): Use upstream entrypoint.sh

### PR DESCRIPTION
**Pull request by jkroepke**
Signed-off-by: Jan-Otto Kröpke <mail@jkroepke.de>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 1]
```diff
@@ -39,6 +39,8 @@ spec:
       {{- end }}
       containers:
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml:40]: Hmmm. Upstream app controller seems to also not use the entrypoint:
https://github.com/argoproj/argo-cd/blob/v2.2.5/manifests/base/application-controller/argocd-application-controller-statefulset.yaml#L21-L23

```yaml
      - command:
        - argocd-application-controller
        env:
```

File: charts/argo-cd/templates/argocd-server/deployment.yaml [hunk 2]
```diff
@@ -41,6 +41,8 @@ spec:
         image: {{ default .Values.global.image.repository .Values.server.image.repository }}:{{ default (include "argo-cd.defaultTag" .) .Values.server.image.tag }}
         imagePullPolicy: {{ default .Values.global.image.imagePullPolicy .Values.server.image.imagePullPolicy }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:42]: Same for api server:
https://github.com/argoproj/argo-cd/blob/v2.2.5/manifests/base/server/argocd-server-deployment.yaml#L23
```yaml
        imagePullPolicy: Always
        command: [argocd-server]
        env:
```

**Review by mkilchhofer — COMMENTED** [review]
While I like having tini (via the entrypoint wrapper) in every pod/component, it seems that only the repo-server uses it in the upstream manifest.

WDYT should we really add it to all components?

**Comment by jkroepke** [comment]
> WDYT should we really add it to all components?

Lets follow the upstream.

**Review by mkilchhofer — APPROVED** [review]
Lgtm 👍

**Pull request merged**